### DISCUSSION TITLE
OTBR docker image, "this is not a bug". What is it then?

### DISCUSSION THREAD

**Discussion by tim-vk** [openthread/openthread#9809] [Q&A] [answered]
I'm not entirely sure if this is the best place to ask this question so please redirect me if i'm at the wrong place.

I'm trying to use the openthread/otbr image in combination with a sonoff-e dongle. However I get the following error:

```
Jan 28 18:25:57 af4a507d35e1 otbr-agent[153]: 51d.18:08:35.950 [D] P-RadioSpinel-: Wait response: tid=1 key=1
Jan 28 18:25:59 af4a507d35e1 otbr-agent[153]: 51d.18:08:37.952 [W] P-RadioSpinel-: Wait for response timeout
Jan 28 18:25:59 af4a507d35e1 otbr-agent[153]: 51d.18:08:37.952 [C] P-RadioSpinel-: Failed to communicate with RCP - no response from RCP during initialization
Jan 28 18:25:59 af4a507d35e1 otbr-agent[153]: 51d.18:08:37.952 [C] P-RadioSpinel-: This is not a bug and typically due a config error (wrong URL parameters) or bad RCP image:
Jan 28 18:25:59 af4a507d35e1 otbr-agent[153]: 51d.18:08:37.952 [C] P-RadioSpinel-: - Make sure RCP is running the correct firmware
Jan 28 18:25:59 af4a507d35e1 otbr-agent[153]: 51d.18:08:37.952 [C] P-RadioSpinel-: - Double check the config parameters passed as `RadioURL` input
```

Full log is attached:
[_openthread_logs (3).txt](https://github.com/openthread/openthread/files/14077355/_openthread_logs.3.txt)

Sure it might be a configuration error but I have no clue what i'm wrongly configuring then.

I've updated the sonoff-e using darkxts web flasher: https://darkxst.github.io/silabs-firmware-builder/ to the latest `Openthread RCP - Thread Only` firmware.

I've pulled the latest docker image of openthread/otbr

i'm runing the docker container with the following settings (i'm using ansible but the syntax is similar enough to docker-compose files):

```
- name : Create the openthread container
  tags: openthread
  community.docker.docker_container:
    name: openthread
    image: openthread/otbr
    restart_policy: unless-stopped
    network_mode: bridge
    privileged: "true"
    sysctls: 
      net.ipv6.conf.all.disable_ipv6: 0 
      net.ipv4.conf.all.forwarding: 1 
      net.ipv6.conf.all.forwarding: 1
    command: --radio-url spinel+hdlc+uart:///dev/ttyACM0
    devices:
      - /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20231122144835-if00:/dev/ttyACM0
    dns_servers:
      - "127.0.0.1"
```
      
   I've verified that the /dev/ path actually exists.
   
   Any clue what i'm doing wrong here? Or where I can potentially can find an answer?

**Comment by tim-vk** [comment, marked as answer]
To whomever reads this:

Apparently you have to add the uart baudrate to the radio url: so `--radio-url spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=460800`.

But i think I've been mostly barking up the wrong tree for the entirety of the solution (I want thread in homeassistant, and all solutions i find don't use this image). 

Godspeed!

**Reply by BacchusIX** [reply]
Sorry, where are you adding this?

**Reply by tim-vk** [reply]
it's a command you have to send to the container on startup.

So:

`docker run --sysctl "net.ipv6.conf.all.disable_ipv6=0 net.ipv4.conf.all.forwarding=1 net.ipv6.conf.all.forwarding=1" -p 8080:80 --dns=127.0.0.1 -it --volume /dev/ttyACM0:/dev/ttyACM0 --privileged openthread/otbr --radio-url spinel+hdlc+uart:///dev/ttyACM0?uart-baudrate=460800`

The --radio-url is the last bit & needs to be complete.
